docs/guide/src/example-plugin/plugin.go

```diff
@@ -6,55 +6,115 @@ import (
 	"github.com/tendermint/go-wire"
 )
 
+//-----------------------------------------
+// Structs
+//   * Note the fields in each struct may be expanded/modified
+
+// Plugin State Struct
+//   * Intended to store the current state of the plugin
+//     * This example contains a field which holds the execution count
+//   * Used by go-wire as the encoding/decoding struct to hold the plugin state
```

Comment:
Sounds redundant with last point. Maybe combine this with The KVStore one: `* Serialized (by go-wire) and stored in the Merkle tree under the key returned by ExamplePlugin.StateKey()`